### PR TITLE
Signup and Login Fixes

### PR DESCRIPTION
**Pull request by aditiii23**

### CODE REVIEW

File: backend/controllers/userController.js [hunk 1]
```diff
@@ -63,6 +63,8 @@ const loginUser = async (req, res, next) => {
         message: "User login successfully",
       })
     }
+    else 
+      throw new ErrorHandler(409, "Wrong Password")
```
Comment (mrpriyansh) [inline backend/controllers/userController.js:67]: The backend changes should be in different pr

File: backend/server.js [hunk 2]
```diff
@@ -16,6 +16,18 @@ connectDB()
 
 const app = express()
 app.use(cors())
+app.use(function (req, res, next) {
```
Comment (mrpriyansh) [inline backend/server.js]: Remove this

File: frontend/src/components/Login/Login.js [hunk 3]
```diff
@@ -36,19 +38,28 @@ const Login = () => {
...
-        navigate("/profile")
-      } else {
-        throw new Error("Invalid login")
+      if (Object.keys(err).length < 1) {
+        let res = await axios.post(`${apiUrl}/users/login`, user)
+        if (res.data.user && res.data.token) {
+          localStorage.setItem("user", JSON.stringify(res.data.user))
+          localStorage.setItem("token", res.data.token)
+          navigate("/profile")
+        } else {
```
Comment (mrpriyansh) [inline frontend/src/components/Login/Login.js]: Is this really required? What's the intention behind this?

File: frontend/src/components/Login/Login.js [hunk 4]
```diff
@@ -36,19 +38,28 @@ const Login = () => {
...
+          return error
+        }
       }
     } catch (err) {
-      if (err.response) {
+      let res = err?.response?.data
+      if (res.message === "No user exists! Please Register") {
```
Comment (mrpriyansh) [inline frontend/src/components/Login/Login.js]: toast.error(res.message) 
Can't we write this?

File: frontend/src/components/Login/Login.js [hunk 5]
```diff
@@ -1,4 +1,6 @@
-import React, { useState, useEffect } from "react"
+import React, { useState } from "react"
+import { toast, ToastContainer } from "react-toastify"
+import "react-toastify/dist/ReactToastify.css"
```
Comment (mrpriyansh) [inline frontend/src/components/Login/Login.js]: Import this in App.js

File: frontend/src/components/Login/Login.js [hunk 6]
```diff
@@ -82,6 +93,7 @@ const Login = () => {
       </form>
       {error?.length > 0 && <div>{error}</div>}
```
Comment (mrpriyansh) [inline frontend/src/components/Login/Login.js]: Remove this

File: frontend/src/components/Login/Login.js [hunk 7]
```diff
@@ -82,6 +93,7 @@ const Login = () => {
...
       <NavLink to="/register">Not yet registered? Register Now</NavLink>
+      <ToastContainer />
```
Comment (mrpriyansh) [inline frontend/src/components/Login/Login.js]: Move this to App.js

File: frontend/src/components/Signup/Signup.js [hunk 8]
```diff
@@ -1,4 +1,6 @@
-import React, { useEffect, useState } from "react"
+import React, { useState } from "react"
+import { toast, ToastContainer } from "react-toastify"
+import "react-toastify/dist/ReactToastify.css"
 import basestyle from "../Base.module.css"
```
Comment (mrpriyansh) [inline frontend/src/components/Signup/Signup.js:3]: Well done.

File: frontend/src/components/Signup/Signup.js [hunk 9]
```diff
@@ -50,26 +53,28 @@ const Signup = () => {
...
+      if (Object.keys(err).length < 1) {
+        const res = await axios.post(`${apiUrl}/users/registerUser/`, user)
+        if (res.data.success) {
+          localStorage.setItem("user", JSON.stringify(res.data.user))
+          navigate("/profile")
+        }
       }
     } catch (err) {
-      if (err.response) {
+      let res = err?.response?.data
```
Comment (mrpriyansh) [inline frontend/src/components/Signup/Signup.js]: res should be const

File: frontend/src/components/Login/Login.js (hunk 3 above)
Comment (mrpriyansh) [reply frontend/src/components/Login/Login.js]: setUserDetails((prevState)=> return {...prevState, [name]: user]})

File: backend/server.js (hunk 2 above)
Comment (aditiii23) [reply backend/server.js]: done

File: frontend/src/components/Login/Login.js (hunk 4 above)
Comment (aditiii23) [reply frontend/src/components/Login/Login.js]: yes, updated

File: frontend/src/components/Login/Login.js (hunk 5 above)
Comment (aditiii23) [reply frontend/src/components/Login/Login.js]: done.

File: frontend/src/components/Login/Login.js (hunk 7 above)
Comment (aditiii23) [reply frontend/src/components/Login/Login.js]: done.

File: frontend/src/components/Signup/Signup.js (hunk 9 above)
Comment (aditiii23) [reply frontend/src/components/Signup/Signup.js]: yes, done

File: frontend/src/components/Login/Login.js (hunk 6 above)
Comment (aditiii23) [reply frontend/src/components/Login/Login.js]: okay

File: frontend/src/components/Login/Login.js (hunk 3 above)
Comment (aditiii23) [reply frontend/src/components/Login/Login.js]: updated

**Review by mrpriyansh — APPROVED** [review]
LGTM
